win_etw_macros/src/lib.rs

```diff
@@ -803,7 +879,7 @@ fn parse_event_field(
                 let field_len = ident_suffix(field_name, "len");
                 statements.extend(quote! {
                     let #field_len: usize = #field_name.len(); // length in code units
-                    let #field_len: u16 = #field_len.min(0xffff) as u16;
+                    let #field_len: u16 = (#field_len * 2 + 1).min(0xffff) as u16; // length in bytes
                 });
                 data_descriptor_array.extend(quote! {
                     EventDataDescriptor::from(&#field_len),
```

Comment:
Not sure why GH won't let me comment further down... but do we need to do the same fix for `WellKnownType::osstr` ? Or is that length supposed to be in characters, not bytes?

Are there docs for how these are supposed to be layed out?

Reply:
Yeah, another good catch.  I've fixed this.

Unfortunately, there isn't a spec on how these fields are encoded. It's all reverse-engineered from `TraceLoggingProvider.h` in the Windows SDK. Fortunately, because that's a header file, this stuff is guaranteed to be ABI-stable.